src/lib/y2s390/dasds_reader.rb

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2s390/dasd"
+require "yast2/execute"
+require "y2s390/dasds_collection"
+require "y2s390/hwinfo_reader"
+
+Yast.import "Mode"
+
+module Y2S390
+  # This class is reponsible of reading the information about DASD devices
+  class DasdsReader
+    attr_accessor :disks
+
+    # Command for displaying configuration of z Systems DASD devices
+    LIST_CMD = "/sbin/lsdasd".freeze
+
+    # Initializes a collection of DASDs based on the information read using lsdasd output and also
+    # the hardware information (hwinfo).
+    #
+    # @param offline [Boolean] whether it should obtain the offline devices too or not
+    # @param force_probing [Boolean] in case of probing it will fetch the hwinfo if not will use the
+    #   information cached when exist
+    # @return [Array<Y2S390::DasdsCollection>] a collection of Dasds read from the system using a
+    #   comination
+    def list(offline: true, force_probing: false)
+      HwinfoReader.instance.reset if force_probing
+
+      a = dasd_entries(offline: offline).each_with_object([]) do |entry, arr|
+        next unless entry.start_with?(/\d/)
+
+        id, offline, name, _, type, = entry.split(" ")
+        attrs = Yast::Mode.config ? {} : { status: offline, device_name: name, type: type }
+        dasd = Y2S390::Dasd.new(id, **attrs).tap do |d|
+          if Yast::Mode.config
+            d.diag = d.use_diag = use_diag?(d)
+            d.format = false
+          else
+            update_additional_info(d)
+          end
+        end
+
+        arr << dasd
+      end
+
+      Y2S390::DasdsCollection.new(a)
+    end
+
+    def update_info(dasd, extended: false)
+      data = dasd_entries(dasd: dasd).find { |e| e.start_with?(/\d/) }
+      return false if data.to_s.empty?
+
+      _, status, name, _, type, = data.split(" ")
+      dasd.status = status
+      dasd.device_name = name
+      dasd.type = type
+      update_additional_info(dasd) if extended
+    end
```

Comment:
Is being used?

Reply:
Auto-reply: yes, it is:https://github.com/yast/yast-s390/pull/93/files#diff-10db1821b2b97379f01ee490a1cd79f7b24006e36cc77c26bfe802ea477e33e9R388